eda/src/autogluon/eda/analysis/explain.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import List, Optional
+
+import pandas as pd
+import shap
+
+from autogluon.core.constants import REGRESSION
+from autogluon.eda import AnalysisState
+from autogluon.eda.analysis.base import AbstractAnalysis
+
+__all__ = ["ShapAnalysis"]
+
+logger = logging.getLogger(__name__)
+
+
+class _ShapAutogluonWrapper:
+    def __init__(self, predictor, feature_names, target_class=None):
+        self.ag_model = predictor
+        self.feature_names = feature_names
+        self.target_class = target_class
+        if target_class is None and predictor.problem_type != REGRESSION:
+            logging.warning("Since target_class not specified, SHAP will explain predictions for each class")
+
+    def predict_proba(self, X):
+        if isinstance(X, pd.Series):
+            X = X.values.reshape(1, -1)
+        if not isinstance(X, pd.DataFrame):
+            X = pd.DataFrame(X, columns=self.feature_names)
+        if self.ag_model.problem_type == REGRESSION:
+            preds = self.ag_model.predict(X)
+        else:
+            preds = self.ag_model.predict_proba(X)
+        if self.ag_model.problem_type == REGRESSION or self.target_class is None:
+            return preds
+        else:
+            return preds[self.target_class]
+
+
+class ShapAnalysis(AbstractAnalysis):
+    """
+    Perform Shapley values calculation using `shap` package for the given rows.
+
+    Parameters
+    ----------
+    rows: pd.DataFrame,
+        rows to explain
+    baseline_sample: int, default = 100
+        The background dataset size to use for integrating out features. To determine the impact
+        of a feature, that feature is set to "missing" and the change in the model output
+        is observed.
```

Comment:
any guidance on what magnitudes represent what noise level?

Is 100 enough? 1000? Will 10000 differ significantly from 100?

Why choose 100 when I could choose 10000? Is it purely to save compute time?

Will it still work if I set it to 1?

Reply:
There should be enough rows to cancel-out significant variance in the base value 
![image](https://user-images.githubusercontent.com/10080307/224827303-9c9aa03d-1f64-4930-aa9e-9c00bf3f0d9a.png)

1 row should work, but you will have garbage output. More rows -> longer it takes to get the values.

Reply:
> "There should be enough rows to cancel-out significant variance in the base value"

Do we assume the user knows what "enough rows" is? Would it be better to give guidance in the doc-string?